pkg/lib/secrets/secrets.go

```diff
@@ -0,0 +1,81 @@
+package secrets
```

Comment:
Do we need a pkg called `lib`? I was under the impression it is best to avoid generic higher level packages.

Reply:
The functions / code common across controllers will be stored in `lib`

Reply:
Then why not in the `controllers` pkg?